pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -51,6 +52,12 @@ func chunkSizeFor(n, parallelism int) int {
 }
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc) {
-	workqueue.ParallelizeUntil(ctx, p.parallelism, pieces, doWorkPiece, workqueue.WithChunkSize(chunkSizeFor(pieces, p.parallelism)))
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, labelValue string) {
+	doWorkPiesWithMetrics := func(piece int) {
+		metrics.Goroutines.WithLabelValues(labelValue).Inc()
+		defer metrics.Goroutines.WithLabelValues(labelValue).Dec()
+		doWorkPiece(piece)
```

Comment:
I am actually cannot get the point of this metric,  how we can get from this metric? it's inc and then dec, what we expect from this metric finally.

I am not sure if the metric can record the intermediate state, this needs to be confirmed.

Reply:
The metric helps monitor the health of the controller, to make sure that the are no goroutine leaks.